explorer/src/pages/TransactionDetailsPage.tsx

```diff
@@ -75,14 +82,54 @@ function StatusCard({ signature }: Props) {
     [fetchStatus, fetchDetails]
   );
 
+  React.useEffect(() => {
+    return () => {
+      if (autoRefreshTimeout.current) {
+        clearTimeout(autoRefreshTimeout.current);
+      }
+    };
+  }, []);
+
   // Fetch transaction on load
   React.useEffect(() => {
     if (!status && clusterStatus === ClusterStatus.Connected) {
       fetchStatus(signature);
     }
   }, [signature, clusterStatus]); // eslint-disable-line react-hooks/exhaustive-deps
 
-  if (!status || status.status === FetchStatus.Fetching) {
+  React.useEffect(() => {
+    if (!status || !status.data?.info) {
+      return;
+    }
+
+    if (status.status !== FetchStatus.Fetching && autoRefreshInProcess) {
+      setAutoRefreshInProcess(false);
+    }
+
+    if (
+      !autoRefreshTimeout.current &&
+      status.data.info.confirmations !== "max"
+    ) {
+      autoRefreshTimeout.current = setTimeout(() => {
+        autoRefreshTimeout.current = null;
+        refresh(signature);
```

Comment:
It would be better to call `fetchStatus` here because we don't need to fetch details until the transaction is confirmed

Reply:
Oh, I gotcha. I was trying to replace the "refresh" button functionality wholesale, which includes fetchStatus and fetchDetails. I guess the final refresh (once it hits max) would be to include fetchDetails as well.

Reply:
Ah ok, yeah the refresh button wasn't setup correctly, thanks for pointing that out! It should have been only refreshing status if details weren't loaded yet. Once the transaction is confirmed, the details will be auto-fetched here: https://github.com/solana-labs/solana/blob/6be6044cb8fa7d29d6cd855163f9711976848c5f/explorer/src/pages/TransactionDetailsPage.tsx#L279

Reply:
Oh yeah, I forgot that was down in the AccountsCard component! I can remove the incorrect refresh fetchDetails in this PR.